src/rest_vol.c

```diff
@@ -2740,6 +2749,163 @@ RV_copy_object_loc_info_callback(char *HTTP_response, void *callback_data_in, vo
+
+    if (NULL == (parse_tree = yajl_tree_parse(HTTP_response, NULL, 0)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsing JSON failed");
+
+    if (NULL == (key_obj = yajl_tree_get(parse_tree, links_keys, yajl_t_array)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to parse links");
+
+    if (key_obj->u.array.len == 0)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "parsed link array was empty");
+
+    if (index >= key_obj->u.array.len)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "requested link index was out of bounds");
+
+    switch (idx_params->order) {
+        case (H5_ITER_DEC):
+            if (NULL == (link_obj = key_obj->u.array.values[key_obj->u.object.len - 1 - index]))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "selected link was NULL");
+            break;
+
+        case (H5_ITER_NATIVE):
+        case (H5_ITER_INC):
+        case (H5_ITER_N):
+        case (H5_ITER_UNKNOWN):
+        default: {
+            if (NULL == (link_obj = key_obj->u.array.values[index]))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "selected link was NULL");
+            break;
+        }
+    }
+
+    /* Iterate through key/value pairs in link response to find name */
+    for (size_t i = 0; i < link_obj->u.object.len; i++) {
+        curr_key = link_obj->u.object.keys[i];
+
+        if (!strcmp(curr_key, "title"))
+            if (NULL == (parsed_link_name = YAJL_GET_STRING(link_obj->u.object.values[i])))
+                FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to get link name");
+    }
+
+    if (NULL == parsed_link_name)
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "server response didn't contain link name");
+
+    if (NULL == (*link_name = RV_malloc(strlen(parsed_link_name) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to allocate memory for link name");
+
+    if (NULL == (memcpy(*link_name, parsed_link_name, strlen(parsed_link_name) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_PARSEERROR, FAIL, "failed to copy link name");
+
+done:
+    if (parse_tree)
+        yajl_tree_free(parse_tree);
+
+    if (ret_value < 0) {
+        RV_free(*link_name);
+        *link_name = NULL;
+    }
+
+    return ret_value;
+} /* end RV_copy_link_name_by_index() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_copy_attribute_name_by_index
+ *
+ * Purpose:     This callback is used to copy the name of an attribute
+ *              in the server's response by index. It allocates heap memory for the name,
+ *              and returns a pointer to the memory by callback_data_out.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              May, 2023
+ */
+herr_t
+RV_copy_attribute_name_by_index(char *HTTP_response, void *callback_data_in, void *callback_data_out)
```

Comment:
The same comments from `RV_copy_link_name_by_index` seem to apply for this function as well.